### PR TITLE
Action History

### PR DESCRIPTION
**Pull request by Vijar**
This just provides a method on context(`context.getActionHistory`) which returns structured debugging data for the action hierarchy. Everything is behind a context feature flag as well.

I'll provide incremental updates to this feature, including an open source action visualizer.

@redonkulus @mridgway @lingyan @kaesonho

### CODE REVIEW

**Comment by yahoocla** [comment]
Thank you for submitting this pull request, however I do not see a valid CLA on file for you. Before we can merge this request please visit https://yahoocla.herokuapp.com/ and agree to the terms. Thanks! :smile:

**Comment by yahoocla** [comment]
CLA is valid!

File: packages/fluxible/lib/FluxibleContext.js [hunk 1]
```diff
@@ -16,9 +16,10 @@ require('setimmediate');
  * A request or browser-session context
  * @class FluxibleContext
  * @param {Fluxible} app The Fluxible instance used to create the context
+ * @param {Object} [options]
```
Comment (redonkulus) [inline packages/fluxible/lib/FluxibleContext.js]: please define what each option is

File: packages/fluxible/lib/FluxibleContext.js [hunk 2]
```diff
@@ -152,7 +157,7 @@ FluxContext.prototype._initializeDispatcher = function initializeDispatcher() {
  * @param {Function} action The action to be executed to get the name from
  * @returns {Object}
  */
-FluxContext.prototype._createSubActionContext = function createSubActionContext(parentActionContext, action) {
+FluxContext.prototype._createSubActionContext = function createSubActionContext(parentActionContext, action, payload) {
```
Comment (redonkulus) [inline packages/fluxible/lib/FluxibleContext.js]: doc block this

**Comment by redonkulus** [comment]
I like it! We should add docs to the FluxibleContext API docs so people know its there and how to use it. We can possibly add this to the site docs as well.

**Comment by Vijar** [comment]
@redonkulus @mridgway updated.

File: packages/fluxible/utils/callAction.js [hunk 3]
```diff
@@ -13,7 +13,8 @@ require('setimmediate');
  * If done callback supplied, that indicates non-Promise invocation expectation,
  * otherwise, Promise invocation.
  */
-function callAction (actionContext, action, payload, done) {
+function callAction (context, actionContext, action, payload, done) {
```
Comment (lingyan) [inline packages/fluxible/utils/callAction.js]: @Vijar Just curious, is `context` added for accessing `_enableDebug` only, or are you planning to make more use of it later?

Comment (Vijar) [inline packages/fluxible/utils/callAction.js]: @lingyan just for the debug flag...do you think I should just pass in the flag?

Comment (lingyan) [inline packages/fluxible/utils/callAction.js]: It's an internal function. Either way is fine with me.

File: packages/fluxible/utils/callAction.js [hunk 4]
```diff
@@ -13,7 +13,8 @@ require('setimmediate');
...
+    var displayName = action.displayName || action.name;
```
Comment (mridgway) [inline packages/fluxible/utils/callAction.js]: Is this used somewhere?

**Comment by mridgway** [comment]
Ok, so the big question: is it possible for this to be a plugin instead of in core? I think you mentioned that measuring the `executeAction` calls from `componentContext` is not possible right now since it does not have a callback. Is that the only thing that you think would block that?

**Comment by Vijar** [comment]
Closing in favor of #418

**Pull request closed without merging**